### DISCUSSION TITLE
Annotations added to Models in openapi spec?

### DISCUSSION THREAD

**Discussion by cycledriver** [fastapi/fastapi#11133] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import json
from fastapi import FastAPI
from pydantic import BaseModel, Field as PydField
from typing import Annotated
from sqlmodel import Field, Text, SQLModel

app = FastAPI()

class PydModel(BaseModel):
    a: Annotated[str, PydField(sa_type=Text(length=1024))]

class SqlModel(SQLModel):
    a: Annotated[str, Field(sa_type=Text(length=1024))]

@app.get("/")
def read_root() -> PydModel:
    return PydModel(a="123")

@app.get("/sqlmodel")
def sqlmodel_path() -> SqlModel:
    return SqlModel(a="123")

if __name__ == "__main__":
   print(json.dumps(app.openapi() ,indent=2))
```

### Description

I mistakenly used a pydantic Field definition when I meant to use a sqlmodel.Field definition.  The resulting openapi spec for pydantic field contains sa_type object in the components.schemas.PydModel.a field.  openapi-generator doesn't like this value here.  Not sure if it's openapi-generator that has a problem or if fastapi is generating an invalid openapi 3.0 spec.

This is the openapi spec generated:

```
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "get": {
        "summary": "Read Root",
        "operationId": "read_root__get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/PydModel"
                }
              }
            }
          }
        }
      }
    },
    "/sqlmodel": {
      "get": {
        "summary": "Sqlmodel Path",
        "operationId": "sqlmodel_path_sqlmodel_get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/SqlModel"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "PydModel": {
        "properties": {
          "a": {
            "type": "string",
            "title": "A",
            "sa_type": {             # Unexpected here!
              "length": 1024,
              "_expect_unicode": false,
              "_warn_on_bytestring": false
            }
          }
        },
        "type": "object",
        "required": [
          "a"
        ],
        "title": "PydModel"
      },
      "SqlModel": {
        "properties": {
          "a": {
            "type": "string",
            "title": "A"
          }
        },
        "type": "object",
        "required": [
          "a"
        ],
        "title": "SqlModel"
      }
    }
  }
}
```

You can see that the SqlModel Field method hides the sa_type from being displayed but the pydantic one passes it through. 

For my code, I really meant to use the SqlModel FIeld so I will switch to that.  

However, I wasn't sure if fastapi is doing the right thing with the pydantic.Field definition from an openapi spec perspective.  Thoughts?  Is this a bug to raise or an expected behaviour.

### Operating System

Linux, Windows, macOS

### FastAPI Version

0.104.1

### Pydantic Version

1.10.14

### Python Version

3.11.6

### Additional Context

With pydantic 2, it refuses to generate the spec with the following:

pydantic_core._pydantic_core.PydanticSerializationError: Unable to serialize unknown type: <class 'sqlalchemy.sql.sqltypes.Text'>

So, I guess this would have prevented me from making the mistake in the first place.

**Comment by YuriiMotov** [comment]
This is not related to FastAPI, but to Pydantic.

Pydantic treats all unknown arguments passed to `Field` as extra and just adds them to schema.
